test/server.test.js

```diff
@@ -1,14 +1,262 @@
+          expect(res.body).to.have.property('token');
+          expect(res.body).to.have.property('redirectUrl').that.equals('/natalia');
+          authToken = res.body.token;
+          done();
+        });
+      });
+  });
+  
+  describe('Users', () =>{
+    describe('GETS /users', () => {
+      it('it should return names of users', (done) => {
+        if (!authToken) return done(new Error('Authentication token did not stick'));
+
+         chai.request(server)
+          .get('/users')
+          .set('Authorization', `Bearer ${authToken}`)
+          .end((err, res) => {
+            if (err) return done(err);
+  
+            expect(res).to.have.status(200);
+            expect(res.body).to.be.an('object');
+
+            const userNames = res.body['users'];
+
+            expect(userNames).to.be.an('array');
+            expect(userNames).to.have.lengthOf(3);
+
+            done();
+          });
+      });
+    });
+  });
+  
+  describe('Cart', () => {
+    describe('GETS /:name', () => {
+      it('it should return existing user\'s cart with product details', (done) => {
+        if (!authToken) return done(new Error('Authentication token did not stick'));
+
+        chai.request(server)
+          .get('/natalia')
+          .set('Authorization', `Bearer ${authToken}`)
+          .end((err, res)=> {
+            if (err) return done(err);
+
+            console.log('Response body:', res.body); 
+  
+            expect(res).to.have.status(200);
+            expect(res.body).to.be.an('object');
+            expect(res.body).to.have.property('userCart').that.is.an('object');
+
+            const userCart = res.body.userCart;
+
+            const item = userCart.items[0];
+            expect(item).to.have.property('product').that.is.a('string');
+            expect(item).to.have.property('quantity');
+            expect(item).to.have.property('price');
+            done();
+          });
+      });
+    });
+
+    describe('POST /:name', () => {
+      it('it should return updated cart of existing user', (done) => {
+        if (!authToken) return done(new Error('Authentication token did not stick'));
+
+        chai.request(server)
+          .post('/natalia')
+          .set('Authorization', `Bearer ${authToken}`)
+          .send({
+              items: [{
+              product: 'glas',
+              quantitty: 1,
+              price: 50
+            }],
+            total: 50
+          })
+          .end((err, res) => {
+            if (err) return done(err);
+  
+            expect(res).to.have.status(200);
+
+            const userCart = res.body['userCart'];
+
+            expect(userCart).to.be.an('object');
+            expect(userCart).to.have.property('items').that.is.an('array');
+            expect(userCart.items[0]).to.have.property('product');
+            expect(userCart.items[1]).to.have.property('quantity');
+            expect(userCart).to.have.property('total').that.is.a('number');
```

Comment:
why not testing the actual number so you test the logic you did to update the total?
```suggestion
            expect(userCart).to.have.property('total').that.equals(sumNumber);
```